src/electrum_wallet_file.rs

```diff
@@ -0,0 +1,438 @@
+                .collect::<Vec<_>>()
+        });
+        if let Some([kind, x]) = captures.as_deref() {
+            let kind = match *kind {
+                "wsh" => "wsh",
+                "sh" => "pkh",
+                "sh(wsh" => "sh(wsh",
+                _ => return Err(format!("unknown nultisig kind: {}", kind)),
+            };
+            let re = Regex::new(r#"[tx]p[ur][bv][0-9A-Za-z]+"#).map_err(|e| e.to_string())?;
+            let keystores = re
+                .captures_iter(desc)
+                .map(|cap| Keystore::new(kind, &cap[0]))
+                .collect::<Result<Vec<Keystore>, _>>()?;
+            let y = keystores.len();
+            if y < 2 {
+                return Err(
+                    "Multisig with less than two signers doesn't make a lot of sense".to_string(),
+                );
+            }
+
+            let wallet = ElectrumWalletFile {
+                addresses: Addresses::new(),
+                keystores,
+                wallet_type: WalletType::Multisig(x.parse().unwrap(), y as u8),
+            };
+            wallet.validate()?;
+            Ok(wallet)
+        } else {
+            Err(format!(
+                "Unknown multisig descriptor format: {:?}",
+                captures
+            ))
+        }
+    }
+
+    /// Generate output descriptors matching the electrum wallet
+    pub fn to_descriptors(&self) -> Result<Vec<String>, String> {
+        match self.wallet_type {
+            WalletType::Standard => {
+                let exkey = self.keystores[0].get_xkey()?;
+                let desc_ext = exkey.kind().to_string() + "(" + &exkey.xkeystr() + "/0/*)";
+                let desc_chg = exkey.kind().to_string() + "(" + &exkey.xkeystr() + "/1/*)";
+                Ok(vec![desc_ext, desc_chg])
+            }
+            WalletType::Multisig(x, _y) => {
+                let xkeys = self
+                    .keystores
+                    .iter()
+                    .map(|ks| ks.get_xkey())
+                    .collect::<Result<Vec<Box<dyn ElectrumExtendedKey>>, _>>()?;
+                let prefix = match &xkeys[0].kind().to_string() as &str {
+                    "pkh" => "sh",
+                    kind => kind,
+                }
+                .to_string();
+                let prefix = format!("{}(sortedmulti({}", prefix, x);
+
+                let mut desc = xkeys.iter().fold(prefix, |acc, exkey| {
+                    acc + &(",".to_string() + &exkey.xkeystr() + "/0/*")
+                });
+                desc += "))";
+                let opening = desc.matches('(').count();
+                let closing = desc.matches(')').count();
+                if opening > closing {
+                    desc += ")"
+                };
+                let desc_chg = desc.replace("/0/*", "/1/*");
+
+                Ok(vec![desc, desc_chg])
+            }
+        }
+    }
+
+    /// validate the internal structure
+    fn validate(&self) -> Result<(), String> {
```

Comment:
I think we should prevent to instantiate `ElectrumWalletFile` struct (removing the pub) and create a method like `new(...) -> Result<Self, Error>` which performs these internal checks

Reply:
fixed in 53bc997b9ed2a84af604ebde49fe233985992ee6